### DISCUSSION TITLE
write_gatt_char response in always None

### DISCUSSION THREAD

**Discussion by LucaAtWork** [hbldh/bleak#1206] [Q&A] [answered]
Hello, I'm writing a test clinet for a bluetooth GATT server.
In one use-case I have to write to a characteristing and read its response.
So this is my code:
**resp = await client.write_gatt_char(ECHO_UUID, b'~TEST_ECHO', response=True)**
but resp in always None.

Now, I've read [this issue](https://github.com/hbldh/bleak/issues/59) but I'm not sure how that guy managed to solve it. I need to get the response from that write_gatt_char function.

Is there any way to achieve this? 

Thanks a lot

**Comment by dlech** [comment, marked as answer]
[Returning None is the expected behavior.](https://bleak.readthedocs.io/en/latest/api/client.html#bleak.BleakClient.write_gatt_char) If there is an error, it will throw an exception.

What were you expecting it to return?

**Reply by LucaAtWork** [reply]
Hi @dlech 
the gatt server replies to that write with the same string used as input.
So I expect to get b'~TEST_ECHO'.
This is how it is currently working on the Android client (developed by my colleagues).

**Reply by dlech** [reply]
I would assume you need to subscribe to notifications to get the reply.

**Reply by LucaAtWork** [reply]
@dlech you are correct. Thank you for your time!
